stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -282,14 +283,23 @@ struct AsyncTask::PrivateStorage {
   /// Currently one word.
   TaskLocal::Storage Local;
 
+  /// State inside the AsyncTask whose state is only managed by the exclusivity
+  /// runtime in stdlibCore. We zero initialize to provide a safe initial value,
+  /// but actually initialize its bit state to a const global provided by
+  /// libswiftCore so that libswiftCore can control the layout of our initial
+  /// state.
+  uintptr_t ExclusivityAccessSet[2] = {0,0};
```

Comment:
I think this might exceed the reserved space on 32-bit. You may need to expand it there.

Reply:
@mikeash I was talking with @rjmccall at one point and he mentioned that I might be able to use different storage rather than in task itself, but I am not sure exactly what he was talking about.